Illuminance triggers (Labs feature) (#45680)
home-assistant/home-assistant.github.io: source/_triggers/illuminance.crossed_threshold.markdown

---
title: "Illuminance crossed threshold"
trigger: illuminance.crossed_threshold
domain: illuminance
description: "Triggers after one or more illuminance values cross a threshold."
related_triggers:
  - illuminance.changed
  - illuminance.detected
  - illuminance.cleared
---

> The **Illuminance crossed threshold** trigger fires when a light level reading crosses into a zone you define. A patio sensor crossing above 30,000 lx when the sun comes out, a desk sensor dipping below 200 lx as it gets dark in the office, a reading entering a comfortable reading range, or a reading escaping that range are all supported.

Reviewer: I'd drop the "reading entering comfortable reading range" because the two readings are confusing:

```suggestion
The **Illuminance crossed threshold** trigger fires when a light level reading crosses into a zone you define. A patio sensor crossing above 30,000 lx when the sun comes out, a desk sensor dipping below 200 lx as it gets dark in the office, or a reading escaping that range are all supported.
```

{% include integrations/labs_entity_triggers_note.md %}

{% include triggers/ui_header.md %}

To use **Illuminance crossed threshold** in an automation:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation, or select **Create automation** > **Create new automation**.
3. In the **When** section, select **Add trigger**.
4. Select what you want to monitor. Under **By target** (see [Targets](#targets)), pick the area your light sensor is in (like your office or patio). You can also select a device, a specific entity, or a label.
5. From the triggers shown for that target, select **Illuminance crossed threshold**.
6. Under **Threshold type**, configure the zone the reading must enter for the trigger to fire:
   - Select **Above** or **Below** and enter a value to fire when the reading crosses that level.
   - Select **In range** and enter a lower and upper bound to fire when the reading enters the range from outside.
   - Select **Outside range** and enter a lower and upper bound to fire when the reading leaves the range (crosses past either bound).

> For each option, you can enter a fixed value in lux, pick a sensor entity, or pick a [number helper](/integrations/input_number/) entity as the threshold. If you don't have a number helper, you can create one by selecting **Create a new number helper**.

Reviewer: Needs some more white space:

```suggestion
      For each option, you can enter a fixed value in lux, pick a sensor entity, or pick a [number helper](/integrations/input_number/) entity as the threshold. If you don't have a number helper, you can create one by selecting **Create a new number helper**.
```

For at least:
  description: How long the reading must remain past the threshold before the trigger fires. Useful to avoid triggering on brief light fluctuations. For example, set it to `0:05:00` to fire only after the reading has stayed past the threshold for 5 minutes. Default is `0` (fires immediately).
{% endoptions_ui %}

{% include triggers/yaml_header.md %}

In YAML, **Illuminance crossed threshold** is referred to as `illuminance.crossed_threshold`. A basic example looks like this:

{% example %}
trigger: |
  trigger: illuminance.crossed_threshold
  target:
    entity_id: sensor.office_illuminance
  options:
    threshold:
      type: below
      value:
        number: 200
{% endexample %}

This fires whenever the office illuminance sensor drops below 200 lx.

To fire when the reading leaves a comfortable reading range (escapes above 750 lx or below 300 lx):

{% example %}
trigger: |
  trigger: illuminance.crossed_threshold
  target:
    entity_id:
      - sensor.office_illuminance
      - sensor.study_illuminance
  options:
    threshold:
      type: outside
      value_min:
        number: 300
      value_max:
        number: 750

>     behavior: any

Reviewer: We can drop the default behavior:

```suggestion
```

  required: false
  type: string
  default: any
for:
  description: |
    How long the reading must remain past the threshold before the trigger fires. Accepts a duration string in `HH:MM:SS` format. For example, `00:05:00` fires only after the reading has stayed past the threshold for 5 minutes.
  required: false
  type: string
  default: "00:00:00"
{% endoptions_yaml %}

{% include triggers/targets.md %}

{% include triggers/behavior.md %}

## Good to know

- Illuminance is measured in lux (lx). For reference: a brightly lit office is around 500 lx, indirect daylight is several thousand lx, and direct sunlight can exceed 100,000 lx.
- **Above** and **Below** fire on the crossing moment only. Once the reading is above the threshold, the trigger does not fire again until the reading dips back below it and then crosses above again.
- **In range** (`between`) fires when the reading moves from outside the bounds into the bounds. **Outside range** (`outside`) fires when the reading moves from inside the bounds past either bound.
- Pair this trigger with [Illuminance changed](/triggers/illuminance.changed/) if you also want to react to smaller fluctuations between crossings.

> - Use **For at least** to ride out brief light fluctuations, such as clouds passing or a person briefly shading an outdoor sensor.

Reviewer: "ride out" is too colloquial. Use e.g. "ignore":

```suggestion
- Use **For at least** to ignore brief light fluctuations, such as clouds passing or a person briefly shading an outdoor sensor.
```
